Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTrackingRecHit.cc

```diff
@@ -0,0 +1,256 @@
+        if (nrechitLayerMapS_primary.find(key) == nrechitLayerMapS_primary.end()) {
+          nrechitLayerMapS_primary.emplace(key, 1);
+        } else {
+          nrechitLayerMapS_primary[key] += 1;
+        }
+      }
+      //GetSimHits
+      const Phase2TrackerRecHit1D* rechit = dynamic_cast<const Phase2TrackerRecHit1D*>(hit);
+      const std::vector<SimHitIdpr>& matchedId = associateRecHit.associateHitId(*rechit);
+      const PSimHit* simhitClosest = nullptr;
+      LocalPoint lp = rechit->localPosition();
+      float mind = 1e4;
+      for (const auto& simHitCol : simHits) {
+        for (const auto& simhitIt : *simHitCol) {
+          if (detId.rawId() != simhitIt.detUnitId())
+            continue;
+          for (auto& mId : matchedId) {
+            if (simhitIt.trackId() == mId.first) {
+              float dx = simhitIt.localPosition().x() - lp.x();
+              float dy = simhitIt.localPosition().y() - lp.y();
+              float dist = dx * dx + dy * dy;
+              if (!simhitClosest || dist < mind) {
+                mind = dist;
+                simhitClosest = &simhitIt;
+              }
+            }
+          }
+        }  //end loop over PSimhitcontainers
+      }    //end loop over simHits
+      if (!simhitClosest)
+        continue;
+      fillOTRecHitHistos(
+          simhitClosest, rechit, selectedSimTrackMap, nrechitLayerMapP_primary, nrechitLayerMapS_primary);
+
+    }  //end loop over rechits of a track
+  }    //End loop over tracks
+
+  //fill nRecHits per event
+  //fill nRecHit counter per layer
+  for (auto& lme : nrechitLayerMapP_primary) {
+    layerMEs_[lme.first].numberRecHitsprimary_P->Fill(nrechitLayerMapP_primary[lme.first]);
+  }
+  for (auto& lme : nrechitLayerMapS_primary) {
+    layerMEs_[lme.first].numberRecHitsprimary_S->Fill(nrechitLayerMapS_primary[lme.first]);
+  }
+}
+
+void Phase2OTValidateTrackingRecHit::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  // call the base fillPsetDescription for the plots bookings
+  Phase2OTValidateRecHitBase::fillPSetDescription(desc);
+
+  //for macro-pixel sensors
+  ///////
+  desc.add<edm::InputTag>("SimVertexSource", edm::InputTag("g4SimHits"));
+  desc.add<bool>("associatePixel", false);
+  desc.add<bool>("associateHitbySimTrack", true);
+  desc.add<bool>("Verbosity", false);
+  desc.add<bool>("associateStrip", true);
+  desc.add<edm::InputTag>("phase2TrackerSimLinkSrc", edm::InputTag("simSiPixelDigis", "Tracker"));
+  desc.add<bool>("associateRecoTracks", false);
+  desc.add<edm::InputTag>("pixelSimLinkSrc", edm::InputTag("simSiPixelDigis", "Pixel"));
+  desc.add<bool>("usePhase2Tracker", true);
+  desc.add<edm::InputTag>("OuterTrackerDigiSimLinkSource", edm::InputTag("simSiPixelDigis", "Tracker"));
+  desc.add<edm::InputTag>("OuterTrackerDigiSource", edm::InputTag("mix", "Tracker"));
```

Comment:
@sroychow I am confused, there is:
```C++
 desc.add<edm::InputTag>("phase2TrackerSimLinkSrc", edm::InputTag("simSiPixelDigis", "Tracker"));
```

but also:
```C++
desc.add<edm::InputTag>("OuterTrackerDigiSimLinkSource", edm::InputTag("simSiPixelDigis", "Tracker"));
```
is the latter used at all?

Reply:
not used. removed it.

Reply:
what about the lower line for the digis (are digis used in general here)?

Reply:
You are correct. Removed it from both the OT modules. Thanks!